src/script/descriptor.cpp

```diff
@@ -740,6 +755,17 @@ class PKDescriptor final : public DescriptorImpl
 public:
     PKDescriptor(std::unique_ptr<PubkeyProvider> prov, bool xonly = false) : DescriptorImpl(Vector(std::move(prov)), "pk"), m_xonly(xonly) {}
     bool IsSingleType() const final { return true; }
+
+    std::optional<int64_t> ScriptSize() const override { return 1 + (m_xonly ? 33 : 32) + 1; }
+
+    std::optional<int64_t> MaxSatSize(bool use_max_sig) const override {
+        const auto ecdsa_sig_size = use_max_sig ? 72 : 71;
+        return 1 + (m_xonly ? ecdsa_sig_size : 64);
```

Comment:
In c8b0aeb534edce08366176275d5577c1e309eb6f "descriptor: introduce a method to get the satisfaction size"

This is backwards, should be 64 for `m_xonly`.

```suggestion
        return 1 + (m_xonly ? 64 : ecdsa_sig_size);
```